remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/ProtobufMetadataManagerInterceptor.java

```diff
@@ -472,7 +523,7 @@ private void updateGlobalErrors(InvocationContext ctx, Set<String> errorFiles, l
          cmd = commandsFactory.buildPutKeyValueCommand(ERRORS_KEY_SUFFIX, sb.toString(),
                keyPartitioner.getSegment(ERRORS_KEY_SUFFIX), DEFAULT_METADATA, flagsBitSet);
       }
-      invoker.running().invoke(ctx, cmd);
+      return invoker.running().invokeAsync(ctx, cmd);
```

Comment:
There is an older issue here (also present in the old blocking code).

The global errors key should contain the errors in all the schemas, not just the errors from the latest schema added/updated. So the interceptor can't use a `PutKeyValueCommand`, it should use a `ComputeCommand` or similar to remove the successful keys and add the new error keys to the global errors atomically.

Reply:
Old indeed. Now I remember it. I hope there's a JIRA for it.

Reply:
I created https://issues.redhat.com/browse/ISPN-12631